File: app/src/main/java/com/luizroseiro/testeandroidv2/mainScreen/StatementsInteractor.java
package com.luizroseiro.testeandroidv2.mainScreen;

import android.support.annotation.NonNull;
import android.view.View;
import android.widget.ProgressBar;

import com.luizroseiro.testeandroidv2.MainActivity;
import com.luizroseiro.testeandroidv2.R;
import com.luizroseiro.testeandroidv2.utils.DataService;
import com.luizroseiro.testeandroidv2.utils.Utils;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

interface StatementsInteractorInput {
    void fetchStatements(StatementsRequest statementsRequest, ProgressBar pbStatements);
}

public class StatementsInteractor implements StatementsInteractorInput {

    private StatementsPresenterInput output;

    StatementsInteractor() {
        this.output = new StatementsPresenter();
    }

    @Override
    public void fetchStatements(StatementsRequest statementsRequest, final ProgressBar pbStatements) {
        DataService.fetchStatements(statementsRequest.getId(), new Callback<StatementsResponse>() {
            @Override
            public void onResponse(@NonNull Call<StatementsResponse> call,
                                   @NonNull Response<StatementsResponse> response) {
                pbStatements.setVisibility(View.GONE);
                if (response.isSuccessful() && response.body() != null) {
                    output.presentStatementsMetaData(response.body());
                }
                else
                    Utils.createToast(MainActivity.getContext()
                            .getString(R.string.error_statements));
            }

            @Override
            public void onFailure(@NonNull Call<StatementsResponse> call, @NonNull Throwable t) {
                pbStatements.setVisibility(View.GONE);
                Utils.createToast(MainActivity.getContext().getString(R.string.error_statements));
            }
        });
    }

}

File: TesteSantander/app/src/main/java/br/com/cauejannini/testesantander/statements/StatementsInteractor.kt
package br.com.cauejannini.testesantander.statements

import android.content.Context
import br.com.cauejannini.testesantander.commons.integracao.ApiRepository
import br.com.cauejannini.testesantander.login.UserAccount
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

interface StatementsInteractorInput {
    fun fetchStatements(userAccount: UserAccount?)
}

class StatementsInteractor: StatementsInteractorInput{

    var output: StatementsPresenterInput? = null
    val api = ApiRepository().get()

    override fun fetchStatements(userAccount: UserAccount?) {

        output?.presentUserAccount(userAccount)

        val userId = userAccount?.userId
        if (userId != null) {

            api.getStatements(userId).enqueue(object: Callback<StatementsResponseModel> {

                override fun onFailure(call: Call<StatementsResponseModel>, t: Throwable) {
                    output?.presentStatementsFailed(if (t.message != null) t.message!! else "Erro")
                }

                override fun onResponse(call: Call<StatementsResponseModel>, response: Response<StatementsResponseModel>) {
                    val statementsResponse = response.body()
                    if (statementsResponse != null) {
                        output?.presentStatementsResponse(statementsResponse)
                    } else {
                        output?.presentStatementsFailed(response.message())
                    }
                }
            })

        } else {
            output?.presentStatementsFailed("User id nulo")
        }


    }
}